ui/pages/confirmations/components/simulation-details/useBalanceChanges.ts

```diff
@@ -102,58 +117,43 @@ function getTokenBalanceChanges(
       asset.standard === TokenStandard.ERC20 ? erc20Decimals[asset.address] : 0;
     const amount = getAssetAmount(tokenBc, decimals);
 
-    const fiatRate = tokenFiatRates[tokenBc.address];
-    const fiatAmount = fiatRate
-      ? amount.times(fiatRate).toNumber()
-      : FIAT_UNAVAILABLE;
+    const fiatRate = erc20FiatRates[tokenBc.address];
+    const fiatAmount =
+      fiatRate === undefined
+        ? FIAT_UNAVAILABLE
+        : amount.times(fiatRate).toNumber();
 
     return { asset, amount, fiatAmount };
   });
 }
 
-// Get the exchange rates for converting tokens to the user's fiat currency.
-const getTokenToFiatConversionRates = createSelector(
-  getConversionRate,
-  getTokenExchangeRates,
-  getConfirmationExchangeRates,
-  (nativeToFiat, contractExchangeRates, confirmationExchangeRates) => {
-    const mergedRates = {
-      ...contractExchangeRates,
-      ...confirmationExchangeRates,
-    } as Record<string, number>;
-    return Object.entries(mergedRates).reduce((acc, [key, value]) => {
-      acc[key.toLowerCase()] = nativeToFiat * value;
-      return acc;
-    }, {} as Record<string, number>);
-  },
-);
-
 // Compiles a list of balance changes from simulation data
 export const useBalanceChanges = (
-  simulationData?: Partial<SimulationData>,
+  simulationData: SimulationData | undefined,
 ): { pending: boolean; value: BalanceChange[] } => {
+  const chainId = useSelector(getCurrentChainId);
```

Comment:
No change needed now, but this is one to keep in mind as we become more multi-chain since we may want this to be a separate argument to `SimulationDetails` in future, so we're not coupled to a global network.

Reply:
Ack. Thank you for making me aware!